src/pubsub/mod_pubsub_cache_mnesia.erl

```diff
@@ -0,0 +1,75 @@
+-module(mod_pubsub_cache_mnesia).
+
+-include("pubsub.hrl").
+-include("jlib.hrl").
+
+-export([start/0, stop/0]).
+
+-export([
+         upsert_last_item/5,
+         get_last_item/2,
+         delete_last_item/2]).
+
+%% ------------------------ Backend start/stop ------------------------
+
+-spec start() -> ok.
+start() ->
+    create_table().
+
+-spec stop() -> ok.
+stop() ->
+    ok.
+
+-spec upsert_last_item(ServerHost :: binary(),
+                       Nidx :: mod_pubsub:nodeIdx(),
+                       ItemID :: mod_pubsub:itemId(),
+                       Publisher::jid:ljid(),
+                       Payload::mod_pubsub:payload()) -> ok | {error, Reason :: term()}.
+upsert_last_item(ServerHost, Nidx, ItemId, Publisher, Payload) ->
+    CreatedAt = os:timestamp(),
+    try mnesia:dirty_write(
+        #pubsub_last_item{
+        nodeid = Nidx,
+        itemid = ItemId,
+        creation = {CreatedAt, {Publisher#jid.luser, ServerHost, <<>>}},
+        payload = Payload}
+    ) of
+        ok -> ok
+    catch
+        exit:{aborted, Reason} -> {error, Reason}
+    end.
+
+-spec get_last_item(ServerHost :: binary(),
+                    Nidx :: mod_pubsub:nodeIdx()) -> [mod_pubsub:pubsubLastItem()] | {error, Reason :: term()}.
+get_last_item(_ServerHost, Nidx) ->
+    try mnesia:dirty_read({pubsub_last_item, Nidx}) of
```

Comment:
I wonder what if there is no items in the node yet? I think this is a correct situation and in such a case I think it'd be good to return sth like `{error, no_items}`. Do we have a test covering such a scenario?